trustchain-api/src/api.rs

```diff
@@ -119,5 +127,242 @@ pub trait TrustchainVCAPI {
     }
 }
 
+#[async_trait]
+pub trait TrustchainVPAPI {
+    /// Signs a presentation constructing a verifiable presentation.
+    async fn sign_presentation(
+        presentation: Presentation,
+        did: &str,
+        key_id: Option<&str>,
+        endpoint: &str,
+        linked_data_proof_options: Option<LinkedDataProofOptions>,
+    ) -> Result<Presentation, PresentationError> {
+        let resolver = get_ion_resolver(endpoint);
+        let attestor = IONAttestor::new(did);
+        Ok(attestor
+            .sign_presentation(&presentation, linked_data_proof_options, key_id, &resolver)
+            .await?)
+    }
+    /// Verifies a verifiable presentation.
+    async fn verify_presentation<T: DIDResolver + Send + Sync>(
+        presentation: &Presentation,
+        ldp_options: Option<LinkedDataProofOptions>,
+        root_event_time: Timestamp,
+        verifier: &IONVerifier<T>,
+    ) -> Result<(), PresentationError> {
+        // Check credentials are present in presentation
+        let credentials = presentation
+            .verifiable_credential
+            .as_ref()
+            .ok_or(PresentationError::NoCredentialsPresent)?;
+
+        // TODO: consider concurrency limit (as rate limiting for verifier requests)
+        let limit = Some(5);
+        let ldp_options_vec: Vec<Option<LinkedDataProofOptions>> = (0..credentials.len())
+            .map(|_| ldp_options.clone())
+            .collect();
+
+        // Verify signatures and issuers for each credential included in the presentation
+        stream::iter(credentials.into_iter().zip(ldp_options_vec))
+            .map(Ok)
+            .try_for_each_concurrent(limit, |(credential_or_jwt, ldp_options)| async move {
+                match credential_or_jwt {
+                    CredentialOrJWT::Credential(credential) => TrustchainAPI::verify_credential(
+                        credential,
+                        ldp_options,
+                        root_event_time,
+                        verifier,
+                    )
+                    .await
+                    .map(|_| ())
+                    .map_err(|err| err.into()),
+                    CredentialOrJWT::JWT(jwt) => {
+                        // TODO: add chain verification for JWT credentials.
```

Comment:
I think we can use this [method](https://docs.rs/ssi/0.4.0/ssi/vc/struct.Credential.html#method.decode_verify_jwt) to extract the credential with a subsequent `TrustchainAPI::verify_credential` call?

Reply:
addressed with 24bdd3bb97b62197c7edf901b1998e2d8343ad8a